cli/src/android/update.ts

```diff
@@ -350,6 +350,65 @@ ext {
     join(config.android.cordovaPluginsDirAbs, 'cordova.variables.gradle'),
     cordovaVariables,
   );
+  await kotlinNeededCheck(config, cordovaPlugins, pluginsGradlePath);
+}
+
+async function kotlinNeededCheck(
```

Comment:
since it's called kotlinNeededCheck, I think it should just return true or false and keep the gradle writing in the handleCordovaPluginsGradle function, you can move the check before the current write on line 341 and avoid to read and rewrite
Also, you can add the `'androidx.core:core-ktx:1.6.0'` and `"org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"` dependencies to the `frameworksArray` when needed so you can avoid a second replace of `SUB-PROJECT\sDEPENDENCIES\sSTART`
The core-ktx version should be configurable too.

Reply:
Restructured 👍